src/site/content/en/blog/timer-throttling/index.md

```diff
@@ -0,0 +1,183 @@
+- If you need to react to stage changes in audio/video, [use events like `timeupdate` and `ended`](https://html.spec.whatwg.org/multipage/media.html#mediaevents), or [`requestVideoFrameCallback`](/requestvideoframecallback-rvfc/) if you need to do something with each frame.
+
+There's also [notification triggers](/notification-triggers/) if you want to show a notification at a particular time.
+
+### Animation
+
+Animation is a visual thing, so it shouldn't use CPU time when the page is hidden.
+
+[`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) is much better at scheduling animation work than JavaScript timers. It synchronizes with the refresh rate of the device, ensuring you only get one callback per displayable frame, and you get the maximum amount of time to construct that frame. Also, `requestAnimationFrame` will wait for the page to be visible, so it doesn't use any CPU when the page is hidden.
+
+If you can declare your whole animation up-front, consider using [CSS animations](https://developer.mozilla.org/en-US/docs/Web/CSS/animation) or the [web animations API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API). These have the same advantages as `requestAnimationFrame`, but the browser can perform additional optimizations like automatic compositing, and they're generally easier to use.
+
+If your animation is low-framerate (like a blinking cursor), timers are still the best option right now, but you can combine them with `requestAnimationFrame` to get the best of both worlds:
+
+```js
+function animationInterval(ms, signal, callback) {
+  const start = performance.now();
+
+  function frame(time) {
+    if (signal.aborted) return;
+    callback(time);
+    scheduleFrame();
+  }
+
+  function scheduleFrame() {
+    const elapsed = performance.now() - start;
+    const roundedElapsed = Math.round(elapsed / ms) * ms;
+    const alignedTime = start + roundedElapsed;
+    const offset = roundedElapsed - elapsed;
+    const delay = ms + offset;
+
+    setTimeout(
+      () =>
+        requestAnimationFrame(() => {
+          frame(alignedTime);
+        }),
+      delay,
+    );
+  }
+
+  scheduleFrame();
+}
+```
+
+Usage:
+
+```js
+const controller = new AbortController();
+
+// Create an animation callback every second:
+animationInterval(1000, controller.signal, (time) => {
+  console.log('tick!', time);
+});
+
+// And stop it:
+controller.abort();
+```
+
+## Testing
+
+This change is in Chrome 88+ and [Chrome Canary](https://www.google.com/intl/en_uk/chrome/canary/), but 5 minutes is a long time to wait for the intense throttling to kick in. Instead, you can launch Chrome with the following command line flag:
```

Comment:
This change will likely be enabled to 100% of Chrome 88 Stable, but is currently only enabled at 50% of Chrome 88+ Canary/Dev/Beta to track the performance impact. We should recommend to set the flag explicitly for testing.

Reply:
Done